test/framework/snow.go

```diff
@@ -96,6 +102,95 @@ func (s *Snow) WithProviderUpgrade(fillers ...api.SnowFiller) ClusterE2ETestOpt
+// as well as the right Bottlerocket AMI ID, devices and osFamily for all SnowMachineConfigs. It also sets any
+// necessary machine config default required for BR, like the container volume size.
+func (s *Snow) WithBottlerocket123() api.ClusterConfigFiller {
+	s.t.Helper()
+	return s.withBottlerocketForKubeVersion(anywherev1.Kube123)
+}
+
+// WithBottlerocket124 returns a cluster config filler that sets the kubernetes version of the cluster to 1.24
+// as well as the right Bottlerocket AMI ID, devices and osFamily for all SnowMachineConfigs. It also sets any
+// necessary machine config default required for BR, like the container volume size.
+func (s *Snow) WithBottlerocket124() api.ClusterConfigFiller {
+	s.t.Helper()
+	return s.withBottlerocketForKubeVersion(anywherev1.Kube124)
+}
+
+// WithUbuntu121 returns a cluster config filler that sets the kubernetes version of the cluster to 1.21
+// as well as the right Ubuntu AMI ID, devices and osFamily for all SnowMachineConfigs.
+func (s *Snow) WithUbuntu121() api.ClusterConfigFiller {
+	s.t.Helper()
+	return s.withKubeVersionAndOS(anywherev1.Kube121, anywherev1.Ubuntu)
+}
+
+// WithUbuntu122 returns a cluster config filler that sets the kubernetes version of the cluster to 1.22
+// as well as the right Ubuntu AMI ID, devices and osFamily for all SnowMachineConfigs.
+func (s *Snow) WithUbuntu122() api.ClusterConfigFiller {
+	s.t.Helper()
+	return s.withKubeVersionAndOS(anywherev1.Kube122, anywherev1.Ubuntu)
+}
+
+// WithUbuntu123 returns a cluster config filler that sets the kubernetes version of the cluster to 1.23
+// as well as the right Ubuntu AMI ID, devices and osFamily for all SnowMachineConfigs.
+func (s *Snow) WithUbuntu123() api.ClusterConfigFiller {
+	s.t.Helper()
+	return s.withKubeVersionAndOS(anywherev1.Kube123, anywherev1.Ubuntu)
+}
+
+// WithUbuntu124 returns a cluster config filler that sets the kubernetes version of the cluster to 1.24
+// as well as the right Ubuntu AMI ID, devices and osFamily for all SnowMachineConfigs.
+func (s *Snow) WithUbuntu124() api.ClusterConfigFiller {
+	s.t.Helper()
+	return s.withKubeVersionAndOS(anywherev1.Kube124, anywherev1.Ubuntu)
+}
+
+func (s *Snow) withBottlerocketForKubeVersion(kubeVersion anywherev1.KubernetesVersion) api.ClusterConfigFiller {
+	return api.JoinClusterConfigFillers(
+		s.withKubeVersionAndOS(kubeVersion, anywherev1.Bottlerocket),
+		api.SnowToConfigFiller(api.WithChangeForAllSnowMachines(api.WithSnowContainersVolumeSize(100))),
+	)
+}
+
+func (s *Snow) withKubeVersionAndOS(kubeVersion anywherev1.KubernetesVersion, osFamily anywherev1.OSFamily) api.ClusterConfigFiller {
+	envar := fmt.Sprintf("T_SNOW_AMIID_%s_%s", strings.ToUpper(string(osFamily)), strings.ReplaceAll(string(kubeVersion), ".", "_"))
+
+	return api.JoinClusterConfigFillers(
+		api.ClusterToConfigFiller(api.WithKubernetesVersion(kubeVersion)),
+		api.SnowToConfigFiller(
+			s.withAMIIDFromEnvVar(envar),
```

Comment:
we can remove amiID and let CAPAS (in the future EKS-A) to decide which amiID to use for most test cases.

Reply:
I left the option to still specify an env var, but I made it optional. If the env var is not set, it will default to empty. That way, we can override it if necessary during manual testing.